internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go

```diff
@@ -0,0 +1,160 @@
+package basicauth
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"net/http"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/util"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/config"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/api"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/auth/provider"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy/apphandler/authproxy/selector"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy/apphandler/chain"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/proxy/pagewriter"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	providerCookie  = "_oauth2_provider"
+	basicAuthCookie = "proxyBasicAuth"
+	formPagePath    = config.InternalPrefix + "/form"
+)
+
+type Handler struct {
+	basicAuth  provider.Basic
+	logger     log.Logger
+	pageWriter *pagewriter.Writer
+	clock      clock.Clock
+	app        api.AppConfig
+	upstream   chain.Handler
+}
+
+func NewHandler(
+	auth provider.Basic,
+	logger log.Logger,
+	pageWriter *pagewriter.Writer,
+	clock clock.Clock,
+	app api.AppConfig,
+	upstream chain.Handler,
+) *Handler {
+	return &Handler{
+		basicAuth:  auth,
+		logger:     logger,
+		app:        app,
+		upstream:   upstream,
+		pageWriter: pageWriter,
+		clock:      clock,
+	}
+}
+
+func (h *Handler) Name() string {
+	return h.basicAuth.Name()
+}
+
+func (h *Handler) SignInPath() string {
+	return formPagePath
+}
+
+func (h *Handler) CookieExpiration() time.Duration {
+	return 1 * time.Hour
+}
+
+func (h *Handler) ServeHTTPOrError(w http.ResponseWriter, req *http.Request) error {
+	host, _ := util.SplitHostPort(req.Host)
+	if host == "" {
+		panic(errors.New("host cannot be empty"))
+	}
+
+	if err := req.ParseForm(); err != nil {
+		return err
+	}
+
+	// Unset content length as we do not want to push req.Body to upstream
+	req.ContentLength = 0
```

Comment:
We do not want to push password that was read from `Form` to upstream. Set ContentLength to 0 to prevent `502` bad gateway error.